Microceph unit's considered Active while not joined to the cluster

While looking at another issue, I've noticed that microceph/3 was considered active, while it did not join the cluster:
```
microceph                      blocked      4  microceph             squid/edge   112

microceph/0*             active    executing  0        10.241.32.116             
microceph/1              blocked   idle       1        10.241.37.82              (workload) Error in charm (see logs): Command '['microceph', 'cluster', 'join', 'eyJzZWNyZXQiOiJlZTU3Yzg5Y2E4YTVjZWZj...
microceph/2              active    idle       2        10.241.32.141             
microceph/3              active    idle       6        10.241.37.32              
```

```
2025-04-27 03:14:11 INFO juju unit_agent.go:288 Starting unit workers for "microceph/3"
2025-04-27 03:14:12 INFO juju.worker.apicaller connect.go:163 [e8b0f6] "unit-microceph-3" successfully connected to "wss://10.241.37.130:17070"
2025-04-27 03:14:12 INFO juju.worker.apicaller connect.go:260 [e8b0f6] password changed for "unit-microceph-3"
2025-04-27 03:14:12 INFO juju.worker.apicaller connect.go:163 [e8b0f6] "unit-microceph-3" successfully connected to "wss://10.241.37.130:17070"
2025-04-27 03:14:12 INFO juju.worker.upgrader upgrader.go:141 no waiter, upgrader is done
2025-04-27 03:14:12 INFO juju.worker.migrationminion worker.go:208 migration phase is now: NONE
2025-04-27 03:14:12 INFO juju.worker.logger logger.go:120 logger worker started
2025-04-27 03:14:12 ERROR juju.worker.meterstatus runner.go:93 error running "meter-status-changed": charm missing from disk
2025-04-27 03:14:12 INFO juju.worker.uniter uniter.go:365 unit "microceph/3" started
2025-04-27 03:14:12 INFO juju.worker.uniter uniter.go:691 resuming charm install
2025-04-27 03:14:12 INFO juju.worker.uniter.charm bundles.go:82 downloading ch:amd64/microceph-112 from API server
2025-04-27 03:14:14 INFO juju.worker.uniter uniter.go:391 hooks are retried true
2025-04-27 03:14:14 INFO juju.worker.uniter.storage resolver.go:126 initial storage attachments ready
2025-04-27 03:14:14 INFO juju.worker.uniter resolver.go:185 found queued "install" hook
2025-04-27 03:14:15 INFO unit.microceph/3.juju-log server.go:406 Running legacy hooks/install.
2025-04-27 03:14:22 INFO juju.worker.uniter.operation runhook.go:186 ran "install" hook (via hook dispatching script: dispatch)
2025-04-27 03:14:29 INFO juju.worker.uniter.operation runhook.go:186 ran "ceph-relation-created" hook (via hook dispatching script: dispatch)
2025-04-27 03:14:36 INFO juju.worker.uniter.operation runhook.go:186 ran "identity-service-relation-created" hook (via hook dispatching script: dispatch)
2025-04-27 03:14:44 INFO juju.worker.uniter.operation runhook.go:186 ran "ceph-relation-created" hook (via hook dispatching script: dispatch)
2025-04-27 03:14:51 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-created" hook (via hook dispatching script: dispatch)
2025-04-27 03:14:58 INFO juju.worker.uniter.operation runhook.go:186 ran "receive-ca-cert-relation-created" hook (via hook dispatching script: dispatch)
2025-04-27 03:15:05 INFO juju.worker.uniter.operation runhook.go:186 ran "traefik-route-rgw-relation-created" hook (via hook dispatching script: dispatch)
2025-04-27 03:15:12 INFO juju.worker.uniter.operation runhook.go:186 ran "leader-settings-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:15:19 INFO unit.microceph/3.juju-log server.go:406 Entering guarded section: 'Checking configs'
2025-04-27 03:15:19 INFO unit.microceph/3.juju-log server.go:406 Entering guarded section: 'Bootstrapping'
2025-04-27 03:15:37 INFO unit.microceph/3.juju-log server.go:406 Setting active status
2025-04-27 03:15:37 INFO unit.microceph/3.juju-log server.go:406 Completed guarded section fully: 'Bootstrapping'
2025-04-27 03:15:37 INFO unit.microceph/3.juju-log server.go:406 Completed guarded section fully: 'Checking configs'
2025-04-27 03:15:38 INFO juju.worker.uniter.operation runhook.go:186 ran "config-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:15:38 INFO juju.worker.uniter resolver.go:185 found queued "start" hook
2025-04-27 03:15:38 INFO unit.microceph/3.juju-log server.go:406 Running legacy hooks/start.
2025-04-27 03:15:45 INFO juju.worker.uniter.operation runhook.go:186 ran "start" hook (via hook dispatching script: dispatch)
2025-04-27 03:15:52 INFO unit.microceph/3.juju-log server.go:406 ceph:8: _on_relation_changed event
2025-04-27 03:15:52 ERROR unit.microceph/3.juju-log server.go:406 ceph:8: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:15:52 WARNING unit.microceph/3.juju-log server.go:406 ceph:8: Microceph not bootstrapped yet.
2025-04-27 03:15:52 INFO unit.microceph/3.juju-log server.go:406 ceph:8: Not processing request as service is not yet ready
2025-04-27 03:15:52 INFO juju.worker.uniter.operation runhook.go:186 ran "ceph-relation-joined" hook (via hook dispatching script: dispatch)
2025-04-27 03:15:59 INFO unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: _on_relation_changed event
2025-04-27 03:15:59 ERROR unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:15:59 WARNING unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Microceph not bootstrapped yet.
2025-04-27 03:15:59 INFO unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Not processing request as service is not yet ready
2025-04-27 03:15:59 INFO juju.worker.uniter.operation runhook.go:186 ran "receive-ca-cert-relation-joined" hook (via hook dispatching script: dispatch)
2025-04-27 03:16:06 INFO unit.microceph/3.juju-log server.go:406 ceph:5: _on_relation_changed event
2025-04-27 03:16:06 ERROR unit.microceph/3.juju-log server.go:406 ceph:5: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:06 WARNING unit.microceph/3.juju-log server.go:406 ceph:5: Microceph not bootstrapped yet.
2025-04-27 03:16:06 INFO unit.microceph/3.juju-log server.go:406 ceph:5: Not processing request as service is not yet ready
2025-04-27 03:16:06 INFO unit.microceph/3.juju-log server.go:406 ceph:5: _on_relation_changed event
2025-04-27 03:16:06 ERROR unit.microceph/3.juju-log server.go:406 ceph:5: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:06 WARNING unit.microceph/3.juju-log server.go:406 ceph:5: Microceph not bootstrapped yet.
2025-04-27 03:16:06 INFO unit.microceph/3.juju-log server.go:406 ceph:5: Not processing request as service is not yet ready
2025-04-27 03:16:06 INFO juju.worker.uniter.operation runhook.go:186 ran "ceph-relation-joined" hook (via hook dispatching script: dispatch)
2025-04-27 03:16:13 INFO unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: _on_relation_changed event
2025-04-27 03:16:13 ERROR unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:13 WARNING unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Microceph not bootstrapped yet.
2025-04-27 03:16:13 INFO unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Not processing request as service is not yet ready
2025-04-27 03:16:13 INFO unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: _on_relation_changed event
2025-04-27 03:16:13 ERROR unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:13 WARNING unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Microceph not bootstrapped yet.
2025-04-27 03:16:13 INFO unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Not processing request as service is not yet ready
2025-04-27 03:16:13 INFO unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Entering guarded section: 'Bootstrapping'
2025-04-27 03:16:32 INFO unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Setting active status
2025-04-27 03:16:32 INFO unit.microceph/3.juju-log server.go:406 receive-ca-cert:11: Completed guarded section fully: 'Bootstrapping'
2025-04-27 03:16:32 INFO juju.worker.uniter.operation runhook.go:186 ran "receive-ca-cert-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:16:38 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: _on_relation_changed event
2025-04-27 03:16:38 ERROR unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:38 WARNING unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Microceph not bootstrapped yet.
2025-04-27 03:16:38 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Not processing request as service is not yet ready
2025-04-27 03:16:39 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: _on_relation_changed event
2025-04-27 03:16:39 ERROR unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:39 WARNING unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Microceph not bootstrapped yet.
2025-04-27 03:16:39 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Not processing request as service is not yet ready
2025-04-27 03:16:39 INFO juju.worker.uniter.operation runhook.go:186 ran "traefik-route-rgw-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:16:46 INFO unit.microceph/3.juju-log server.go:406 ceph:8: _on_relation_changed event
2025-04-27 03:16:46 ERROR unit.microceph/3.juju-log server.go:406 ceph:8: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:46 WARNING unit.microceph/3.juju-log server.go:406 ceph:8: Microceph not bootstrapped yet.
2025-04-27 03:16:46 INFO unit.microceph/3.juju-log server.go:406 ceph:8: Not processing request as service is not yet ready
2025-04-27 03:16:46 INFO unit.microceph/3.juju-log server.go:406 ceph:8: _on_relation_changed event
2025-04-27 03:16:46 ERROR unit.microceph/3.juju-log server.go:406 ceph:8: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:46 WARNING unit.microceph/3.juju-log server.go:406 ceph:8: Microceph not bootstrapped yet.
2025-04-27 03:16:46 INFO unit.microceph/3.juju-log server.go:406 ceph:8: Not processing request as service is not yet ready
2025-04-27 03:16:46 INFO unit.microceph/3.juju-log server.go:406 ceph:8: _on_relation_changed event
2025-04-27 03:16:46 ERROR unit.microceph/3.juju-log server.go:406 ceph:8: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:46 WARNING unit.microceph/3.juju-log server.go:406 ceph:8: Microceph not bootstrapped yet.
2025-04-27 03:16:46 INFO unit.microceph/3.juju-log server.go:406 ceph:8: Not processing request as service is not yet ready
2025-04-27 03:16:46 INFO juju.worker.uniter.operation runhook.go:186 ran "ceph-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:16:53 INFO unit.microceph/3.juju-log server.go:406 ceph:5: _on_relation_changed event
2025-04-27 03:16:53 ERROR unit.microceph/3.juju-log server.go:406 ceph:5: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:53 WARNING unit.microceph/3.juju-log server.go:406 ceph:5: Microceph not bootstrapped yet.
2025-04-27 03:16:53 INFO unit.microceph/3.juju-log server.go:406 ceph:5: Not processing request as service is not yet ready
2025-04-27 03:16:53 INFO unit.microceph/3.juju-log server.go:406 ceph:5: _on_relation_changed event
2025-04-27 03:16:53 ERROR unit.microceph/3.juju-log server.go:406 ceph:5: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:53 WARNING unit.microceph/3.juju-log server.go:406 ceph:5: Microceph not bootstrapped yet.
2025-04-27 03:16:53 INFO unit.microceph/3.juju-log server.go:406 ceph:5: Not processing request as service is not yet ready
2025-04-27 03:16:53 INFO unit.microceph/3.juju-log server.go:406 ceph:5: _on_relation_changed event
2025-04-27 03:16:53 ERROR unit.microceph/3.juju-log server.go:406 ceph:5: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:53 WARNING unit.microceph/3.juju-log server.go:406 ceph:5: Microceph not bootstrapped yet.
2025-04-27 03:16:53 INFO unit.microceph/3.juju-log server.go:406 ceph:5: Not processing request as service is not yet ready
2025-04-27 03:16:53 INFO unit.microceph/3.juju-log server.go:406 ceph:5: _on_relation_changed event
2025-04-27 03:16:53 ERROR unit.microceph/3.juju-log server.go:406 ceph:5: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:16:53 WARNING unit.microceph/3.juju-log server.go:406 ceph:5: Microceph not bootstrapped yet.
2025-04-27 03:16:53 INFO unit.microceph/3.juju-log server.go:406 ceph:5: Not processing request as service is not yet ready
2025-04-27 03:16:54 INFO juju.worker.uniter.operation runhook.go:186 ran "ceph-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:17:00 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: _on_relation_changed event
2025-04-27 03:17:00 ERROR unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:00 WARNING unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Microceph not bootstrapped yet.
2025-04-27 03:17:00 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Not processing request as service is not yet ready
2025-04-27 03:17:00 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: _on_relation_changed event
2025-04-27 03:17:00 ERROR unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:00 WARNING unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Microceph not bootstrapped yet.
2025-04-27 03:17:00 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Not processing request as service is not yet ready
2025-04-27 03:17:01 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: _on_relation_changed event
2025-04-27 03:17:01 ERROR unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:01 WARNING unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Microceph not bootstrapped yet.
2025-04-27 03:17:01 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Not processing request as service is not yet ready
2025-04-27 03:17:01 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: _on_relation_changed event
2025-04-27 03:17:01 ERROR unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:01 WARNING unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Microceph not bootstrapped yet.
2025-04-27 03:17:01 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Not processing request as service is not yet ready
2025-04-27 03:17:01 INFO juju.worker.uniter.operation runhook.go:186 ran "traefik-route-rgw-relation-joined" hook (via hook dispatching script: dispatch)
2025-04-27 03:17:08 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:08 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:08 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:08 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:08 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:08 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:08 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:08 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:08 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:08 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:08 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:08 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:08 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:08 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:08 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:08 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:14 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Entering guarded section: 'Bootstrapping'
2025-04-27 03:17:33 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Setting active status
2025-04-27 03:17:33 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Completed guarded section fully: 'Bootstrapping'
2025-04-27 03:17:33 INFO juju.worker.uniter.operation runhook.go:186 ran "identity-service-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:17:40 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: _on_relation_changed event
2025-04-27 03:17:40 ERROR unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:40 WARNING unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Microceph not bootstrapped yet.
2025-04-27 03:17:40 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Not processing request as service is not yet ready
2025-04-27 03:17:40 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: _on_relation_changed event
2025-04-27 03:17:40 ERROR unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:40 WARNING unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Microceph not bootstrapped yet.
2025-04-27 03:17:40 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Not processing request as service is not yet ready
2025-04-27 03:17:40 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: _on_relation_changed event
2025-04-27 03:17:40 ERROR unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:40 WARNING unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Microceph not bootstrapped yet.
2025-04-27 03:17:40 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Not processing request as service is not yet ready
2025-04-27 03:17:40 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: _on_relation_changed event
2025-04-27 03:17:40 ERROR unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:40 WARNING unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Microceph not bootstrapped yet.
2025-04-27 03:17:40 INFO unit.microceph/3.juju-log server.go:406 traefik-route-rgw:9: Not processing request as service is not yet ready
2025-04-27 03:17:40 INFO juju.worker.uniter.operation runhook.go:186 ran "traefik-route-rgw-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:17:47 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:47 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:47 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:47 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:47 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:47 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:47 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:47 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:47 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:47 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:47 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:47 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:47 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:47 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:48 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:48 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:48 INFO juju.worker.uniter.operation runhook.go:186 ran "identity-service-relation-joined" hook (via hook dispatching script: dispatch)
2025-04-27 03:17:54 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:55 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:55 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:55 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:55 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:55 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:55 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:55 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:55 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:55 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:55 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:55 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:17:55 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: _on_relation_changed event
2025-04-27 03:17:55 ERROR unit.microceph/3.juju-log server.go:406 identity-service:10: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:17:55 WARNING unit.microceph/3.juju-log server.go:406 identity-service:10: Microceph not bootstrapped yet.
2025-04-27 03:17:55 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Not processing request as service is not yet ready
2025-04-27 03:18:01 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Entering guarded section: 'Bootstrapping'
2025-04-27 03:18:19 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Setting active status
2025-04-27 03:18:19 INFO unit.microceph/3.juju-log server.go:406 identity-service:10: Completed guarded section fully: 'Bootstrapping'
2025-04-27 03:18:20 INFO juju.worker.uniter.operation runhook.go:186 ran "identity-service-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:18:27 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:27 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:27 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:27 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:27 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:27 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:27 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:27 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:28 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:28 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:28 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:28 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:28 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:28 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:28 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:28 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:28 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:18:35 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:35 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:35 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:35 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:35 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:35 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:35 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:35 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:35 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:35 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:35 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:35 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:35 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:35 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:35 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:35 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:35 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:18:42 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:42 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:42 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:42 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:42 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:42 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:42 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:42 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:42 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:42 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:42 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:42 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:42 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:43 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:43 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:43 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:43 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-joined" hook (via hook dispatching script: dispatch)
2025-04-27 03:18:49 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:50 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:50 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:50 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:50 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:50 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:50 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:50 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:50 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:50 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:50 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:50 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:50 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:50 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:50 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:50 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:50 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:18:57 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:57 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:57 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:57 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:57 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:57 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:57 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:57 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:57 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:57 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:57 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:57 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:57 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:18:57 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:18:57 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:18:57 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:18:58 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-joined" hook (via hook dispatching script: dispatch)
2025-04-27 03:19:04 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:04 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:04 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:04 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:04 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:04 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:04 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:04 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:05 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:05 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:05 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:05 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:05 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:05 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:05 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:05 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:05 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:19:12 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:12 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:12 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:12 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:12 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:12 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:12 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:12 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:12 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:12 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:12 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:12 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:12 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:12 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:12 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:12 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:12 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-joined" hook (via hook dispatching script: dispatch)
2025-04-27 03:19:19 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:19 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:19 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:19 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:19 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:19 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:19 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:19 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:19 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:19 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:19 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:19 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:19 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:19 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:19 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:19 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:20 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:19:33 INFO unit.microceph/3.juju-log server.go:406 _on_relation_changed event
2025-04-27 03:19:33 ERROR unit.microceph/3.juju-log server.go:406 Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:33 WARNING unit.microceph/3.juju-log server.go:406 Microceph not bootstrapped yet.
2025-04-27 03:19:33 INFO unit.microceph/3.juju-log server.go:406 Not processing request as service is not yet ready
2025-04-27 03:19:33 INFO unit.microceph/3.juju-log server.go:406 _on_relation_changed event
2025-04-27 03:19:33 ERROR unit.microceph/3.juju-log server.go:406 Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:33 WARNING unit.microceph/3.juju-log server.go:406 Microceph not bootstrapped yet.
2025-04-27 03:19:33 INFO unit.microceph/3.juju-log server.go:406 Not processing request as service is not yet ready
2025-04-27 03:19:33 INFO unit.microceph/3.juju-log server.go:406 _on_relation_changed event
2025-04-27 03:19:34 ERROR unit.microceph/3.juju-log server.go:406 Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:34 WARNING unit.microceph/3.juju-log server.go:406 Microceph not bootstrapped yet.
2025-04-27 03:19:34 INFO unit.microceph/3.juju-log server.go:406 Not processing request as service is not yet ready
2025-04-27 03:19:34 INFO unit.microceph/3.juju-log server.go:406 _on_relation_changed event
2025-04-27 03:19:34 ERROR unit.microceph/3.juju-log server.go:406 Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:34 WARNING unit.microceph/3.juju-log server.go:406 Microceph not bootstrapped yet.
2025-04-27 03:19:34 INFO unit.microceph/3.juju-log server.go:406 Not processing request as service is not yet ready
2025-04-27 03:19:41 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:41 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:41 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:41 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:41 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:41 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:41 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:41 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:41 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:41 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:41 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:41 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:41 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:19:41 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:19:41 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:19:41 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:19:41 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:20:43 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:20:43 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:20:43 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:20:43 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:20:43 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:20:43 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:20:43 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:20:43 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:20:43 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:20:43 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:20:43 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:20:43 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:20:43 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:20:43 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:20:43 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:20:43 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:20:44 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-changed" hook (via hook dispatching script: dispatch)
2025-04-27 03:21:55 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:21:56 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:21:56 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:21:56 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:21:56 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:21:56 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:21:56 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:21:56 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:21:56 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:21:56 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:21:56 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:21:56 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:21:56 INFO unit.microceph/3.juju-log server.go:406 peers:3: _on_relation_changed event
2025-04-27 03:21:56 ERROR unit.microceph/3.juju-log server.go:406 peers:3: Failed executing cmd: ['microceph', 'status'], error: Error: failed listing disks: Database is not yet initialized

2025-04-27 03:21:56 WARNING unit.microceph/3.juju-log server.go:406 peers:3: Microceph not bootstrapped yet.
2025-04-27 03:21:56 INFO unit.microceph/3.juju-log server.go:406 peers:3: Not processing request as service is not yet ready
2025-04-27 03:21:56 INFO juju.worker.uniter.operation runhook.go:186 ran "peers-relation-changed" hook (via hook dispatching script: dispatch)
```

--- a/src/charm.py
+++ b/src/charm.py
@@ -517,7 +517,7 @@ class MicroCephCharm(sunbeam_charm.OSBaseOperatorCharm):
 
             if error_socket_not_exists in e.stderr:
                 event.defer()
-                return
+                raise sunbeam_guard.WaitingExceptionError("waiting for microceph service") 
 
             if error_already_exists not in e.stderr:
                 raise e
@@ -629,7 +629,7 @@ class MicroCephCharm(sunbeam_charm.OSBaseOperatorCharm):
         """Configure Ceph."""
         if not self.ready_for_service():
             event.defer()
-            return
+            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service")
 
         try:
             default_rf = self.model.config.get("default-pool-size")

--- a/src/charm.py
+++ b/src/charm.py
@@ -517,7 +517,7 @@ class MicroCephCharm(sunbeam_charm.OSBaseOperatorCharm):
 
             if error_socket_not_exists in e.stderr:
                 event.defer()
-                raise sunbeam_guard.WaitingExceptionError("waiting for microceph service") 
+                raise sunbeam_guard.WaitingExceptionError("waiting for microceph service")
 
             if error_already_exists not in e.stderr:
                 raise e

--- a/src/cluster.py
+++ b/src/cluster.py
@@ -51,7 +51,7 @@ class ClusterNodes(ops.framework.Object):
         )
         if not hostnames:
             event.defer()
-            return
+            raise sunbeam_guard.WaitingExceptionError("waiting for peer data")
 
         cmd = ["microceph", "cluster", "add", hostnames[0]]
         try:
@@ -76,7 +76,7 @@ class ClusterNodes(ops.framework.Object):
         if not token:
             logger.info("Token not available, deferring join event.")
             event.defer()
-            return
+            raise sunbeam_guard.WaitingExceptionError("waiting for join token") 
 
         if self.charm.peers.interface.state.joined is True:
             logger.info("Unit has already joined the cluster")

--- a/src/cluster.py
+++ b/src/cluster.py
@@ -76,7 +76,7 @@ class ClusterNodes(ops.framework.Object):
         if not token:
             logger.info("Token not available, deferring join event.")
             event.defer()
-            raise sunbeam_guard.WaitingExceptionError("waiting for join token") 
+            raise sunbeam_guard.WaitingExceptionError("waiting for join token")
 
         if self.charm.peers.interface.state.joined is True:
             logger.info("Unit has already joined the cluster")

--- a/src/microceph.py
+++ b/src/microceph.py
@@ -23,8 +23,8 @@ import subprocess
 from socket import gethostname
 from typing import Tuple
 
-import requests
 import ops_sunbeam.guard as sunbeam_guard
+import requests
 from charms.operator_libs_linux.v2 import snap
 from tenacity import retry, stop_after_attempt, wait_fixed
 
@@ -74,7 +74,7 @@ def cos_agent_refresh_cb(event):
         if not is_ready():
             logger.debug("not bootstrapped, defer _on_refresh: %s", event)
             event.defer()
-            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service") 
+            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service")
 
         logger.debug("refreshing cos_agent relation")
         enable_ceph_monitoring()

--- a/src/microceph.py
+++ b/src/microceph.py
@@ -24,6 +24,7 @@ from socket import gethostname
 from typing import Tuple
 
 import requests
+import ops_sunbeam.guard as sunbeam_guard
 from charms.operator_libs_linux.v2 import snap
 from tenacity import retry, stop_after_attempt, wait_fixed
 
@@ -69,15 +70,14 @@ def is_ready() -> bool:
 
 def cos_agent_refresh_cb(event):
     """Callback for cos-agent relation change."""
-    logger.info("Entered CEPH COS AGENT REFRESH")
+    with sunbeam_guard.guard("microceph", "cos_agent"):
+        if not is_ready():
+            logger.debug("not bootstrapped, defer _on_refresh: %s", event)
+            event.defer()
+            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service") 
 
-    if not is_ready():
-        logger.debug("not bootstrapped, defer _on_refresh: %s", event)
-        event.defer()
-        return
-
-    logger.debug("refreshing cos_agent relation")
-    enable_ceph_monitoring()
+        logger.debug("refreshing cos_agent relation")
+        enable_ceph_monitoring()
 
 
 def cos_agent_departed_cb(event):

--- a/src/microceph.py
+++ b/src/microceph.py
@@ -23,7 +23,6 @@ import subprocess
 from socket import gethostname
 from typing import Tuple
 
-import ops_sunbeam.guard as sunbeam_guard
 import requests
 from charms.operator_libs_linux.v2 import snap
 from tenacity import retry, stop_after_attempt, wait_fixed
@@ -70,14 +69,13 @@ def is_ready() -> bool:
 
 def cos_agent_refresh_cb(event):
     """Callback for cos-agent relation change."""
-    with sunbeam_guard.guard("microceph", "cos_agent"):
-        if not is_ready():
-            logger.debug("not bootstrapped, defer _on_refresh: %s", event)
-            event.defer()
-            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service")
+    if not is_ready():
+        logger.debug("not bootstrapped, defer _on_refresh: %s", event)
+        event.defer()
+        return
 
-        logger.debug("refreshing cos_agent relation")
-        enable_ceph_monitoring()
+    logger.debug("refreshing cos_agent relation")
+    enable_ceph_monitoring()
 
 
 def cos_agent_departed_cb(event):

--- a/src/relation_handlers.py
+++ b/src/relation_handlers.py
@@ -23,8 +23,8 @@ import json
 import logging
 from typing import Callable, Dict, List, Optional, Tuple
 
-import requests
 import ops_sunbeam.guard as sunbeam_guard
+import requests
 from ops.charm import CharmBase, RelationEvent
 from ops.framework import EventBase, EventSource, Handle, Object, ObjectEvents, StoredState
 from ops_sunbeam.interfaces import OperatorPeers
@@ -311,7 +311,7 @@ class MicroClusterPeerHandler(BasePeerHandler):
         if not self.is_leader_ready():
             logger.debug("Add node event, deferring the event as leader not ready")
             event.defer()
-            raise sunbeam_guard.WaitingExceptionError("waiting for charm leader") 
+            raise sunbeam_guard.WaitingExceptionError("waiting for charm leader")
 
         if not self.model.unit.is_leader():
             logger.debug("Ignoring Add node event as this is not leader unit")
@@ -332,7 +332,7 @@ class MicroClusterPeerHandler(BasePeerHandler):
         if not self.is_leader_ready():
             logger.debug("Remove node event, deferring the event as leader not ready")
             event.defer()
-            raise sunbeam_guard.WaitingExceptionError("waiting for charm leader") 
+            raise sunbeam_guard.WaitingExceptionError("waiting for charm leader")
 
         if not self.model.unit.is_leader():
             logger.debug("Ignoring Remove node event as this is not leader unit")
@@ -345,7 +345,7 @@ class MicroClusterPeerHandler(BasePeerHandler):
         if not self.is_leader_ready():
             logger.debug("Upgrade request event, deferring the event as leader not ready")
             event.defer()
-            raise sunbeam_guard.WaitingExceptionError("waiting for charm leader") 
+            raise sunbeam_guard.WaitingExceptionError("waiting for charm leader")
 
         self.upgrade_callback(event)
 
@@ -433,12 +433,12 @@ class CephClientProvides(Object):
         if not self.charm.ready_for_service():
             logger.info("Not processing request as service is not yet ready")
             event.defer()
-            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service") 
+            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service")
 
         if get_osd_count() == 0:
             logger.info("Storage not available, deferring event.")
             event.defer()
-            raise sunbeam_guard.WaitingExceptionError("waiting for storage to be attached") 
+            raise sunbeam_guard.WaitingExceptionError("waiting for storage to be attached")
 
         self._handle_client_relation(event.relation, event.unit)
 
@@ -641,7 +641,7 @@ class CephClientProviderHandler(RelationHandler):
         if not self.can_service(event):
             logger.info("Deferring handling of relation: %s" % self.relation_name)
             event.defer()
-            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service") 
+            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service")
 
         logger.info(f"Processing broker req {event.broker_req}")
         broker_result = process_requests(event.broker_req)

--- a/src/relation_handlers.py
+++ b/src/relation_handlers.py
@@ -24,6 +24,7 @@ import logging
 from typing import Callable, Dict, List, Optional, Tuple
 
 import requests
+import ops_sunbeam.guard as sunbeam_guard
 from ops.charm import CharmBase, RelationEvent
 from ops.framework import EventBase, EventSource, Handle, Object, ObjectEvents, StoredState
 from ops_sunbeam.interfaces import OperatorPeers
@@ -310,7 +311,7 @@ class MicroClusterPeerHandler(BasePeerHandler):
         if not self.is_leader_ready():
             logger.debug("Add node event, deferring the event as leader not ready")
             event.defer()
-            return
+            raise sunbeam_guard.WaitingExceptionError("waiting for charm leader") 
 
         if not self.model.unit.is_leader():
             logger.debug("Ignoring Add node event as this is not leader unit")
@@ -331,7 +332,7 @@ class MicroClusterPeerHandler(BasePeerHandler):
         if not self.is_leader_ready():
             logger.debug("Remove node event, deferring the event as leader not ready")
             event.defer()
-            return
+            raise sunbeam_guard.WaitingExceptionError("waiting for charm leader") 
 
         if not self.model.unit.is_leader():
             logger.debug("Ignoring Remove node event as this is not leader unit")
@@ -344,7 +345,7 @@ class MicroClusterPeerHandler(BasePeerHandler):
         if not self.is_leader_ready():
             logger.debug("Upgrade request event, deferring the event as leader not ready")
             event.defer()
-            return
+            raise sunbeam_guard.WaitingExceptionError("waiting for charm leader") 
 
         self.upgrade_callback(event)
 
@@ -432,12 +433,12 @@ class CephClientProvides(Object):
         if not self.charm.ready_for_service():
             logger.info("Not processing request as service is not yet ready")
             event.defer()
-            return
+            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service") 
 
         if get_osd_count() == 0:
             logger.info("Storage not available, deferring event.")
             event.defer()
-            return
+            raise sunbeam_guard.WaitingExceptionError("waiting for storage to be attached") 
 
         self._handle_client_relation(event.relation, event.unit)
 
@@ -640,7 +641,7 @@ class CephClientProviderHandler(RelationHandler):
         if not self.can_service(event):
             logger.info("Deferring handling of relation: %s" % self.relation_name)
             event.defer()
-            return
+            raise sunbeam_guard.WaitingExceptionError("waiting for microceph service") 
 
         logger.info(f"Processing broker req {event.broker_req}")
         broker_result = process_requests(event.broker_req)

--- a/src/storage.py
+++ b/src/storage.py
@@ -77,24 +77,24 @@ class StorageHandler(Object):
 
     def _on_osd_standalone_attached(self, event: StorageAttachedEvent):
         """Storage attached handler for osd-standalone."""
-        if not self.charm.ready_for_service():
-            logger.warning("MicroCeph not ready yet, deferring storage event.")
-            event.defer()
-            return
-
-        self._clean_stale_osd_data()
-
-        enroll = []
-
-        logger.debug(f"storage on unit: {self._fetch_filtered_storages([self.standalone])}")
-
-        for storage in self._fetch_filtered_storages([self.standalone]):
-            logger.debug(f"Processing {storage}")
-            if not self._get_osd_id(name=storage):
-                enroll.append(storage)
-
-        logger.debug(f"Enroll list {enroll}")
         with sunbeam_guard.guard(self.charm, self.name):
+            if not self.charm.ready_for_service():
+                logger.warning("MicroCeph not ready yet, deferring storage event.")
+                event.defer()
+                raise sunbeam_guard.WaitingExceptionError("waiting for microceph service") 
+
+            self._clean_stale_osd_data()
+
+            enroll = []
+
+            logger.debug(f"storage on unit: {self._fetch_filtered_storages([self.standalone])}")
+
+            for storage in self._fetch_filtered_storages([self.standalone]):
+                logger.debug(f"Processing {storage}")
+                if not self._get_osd_id(name=storage):
+                    enroll.append(storage)
+
+            logger.debug(f"Enroll list {enroll}")
             self.charm.status.set(MaintenanceStatus("Enrolling OSDs"))
             self._enroll_disks_in_batch(enroll)
             self.charm.status.set(ActiveStatus("charm is ready"))

--- a/src/storage.py
+++ b/src/storage.py
@@ -81,7 +81,7 @@ class StorageHandler(Object):
             if not self.charm.ready_for_service():
                 logger.warning("MicroCeph not ready yet, deferring storage event.")
                 event.defer()
-                raise sunbeam_guard.WaitingExceptionError("waiting for microceph service") 
+                raise sunbeam_guard.WaitingExceptionError("waiting for microceph service")
 
             self._clean_stale_osd_data()
 

--- a/tests/unit/test_charm.py
+++ b/tests/unit/test_charm.py
@@ -18,6 +18,7 @@ from pathlib import Path
 from subprocess import CalledProcessError
 from unittest.mock import MagicMock, PropertyMock, call, mock_open, patch
 
+import ops_sunbeam.guard as sunbeam_guard
 import ops_sunbeam.test_utils as test_utils
 from charms.ceph_mon.v0 import ceph_cos_agent
 from ops.testing import Harness
@@ -59,6 +60,9 @@ class _MicroCephCharm(charm.MicroCephCharm):
         super().__init__(framework)
 
     def configure_ceph(self, event):
+        # check if configure_ceph is called from unit tests.
+        if isinstance(event, MagicMock):
+            super().configure_ceph(event)
         return True
 
 
@@ -987,6 +991,14 @@ class TestCharm(test_utils.CharmTestCase):
             }
         )
         action_event.fail.assert_called()
+
+    @patch.object(microceph, "subprocess")
+    @patch("charm.microceph_client.Client")
+    def test_waiting_exception_raised(self, _client, _sub):
+        """Tests whether waiting exception is raised for waiting."""
+        with self.assertRaises(sunbeam_guard.WaitingExceptionError):
+            event = MagicMock()
+            self.harness.charm.configure_ceph(event)
 
     @patch("microceph.is_ready")
     @patch("microceph.enable_mgr_module")

--- a/tests/unit/test_charm.py
+++ b/tests/unit/test_charm.py
@@ -228,7 +228,7 @@ class TestCharm(test_utils.CharmTestCase):
 
         self.harness.set_leader()
         self.harness.update_config({"snap-channel": "1.0/stable", "enable-rgw": "*"})
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
         self.add_complete_identity_relation(self.harness)
         self.add_complete_ingress_relation(self.harness)
         self.add_complete_certificate_transfer_relation(self.harness)
@@ -290,7 +290,7 @@ class TestCharm(test_utils.CharmTestCase):
         self.harness.update_config(
             {"snap-channel": "1.0/stable", "enable-rgw": "*", "namespace-projects": True}
         )
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
         self.add_complete_identity_relation(self.harness)
         self.add_complete_ingress_relation(self.harness)
         self.add_complete_certificate_transfer_relation(self.harness)
@@ -353,7 +353,7 @@ class TestCharm(test_utils.CharmTestCase):
         self.harness.update_config(
             {"snap-channel": "1.0/stable", "enable-rgw": "*", "namespace-projects": True}
         )
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
         self.add_complete_identity_relation(self.harness)
         self.add_complete_ingress_relation(self.harness)
         subprocess.run.assert_any_call(
@@ -401,7 +401,7 @@ class TestCharm(test_utils.CharmTestCase):
     @patch("ceph.check_output")
     def test_add_osds_action_with_device_id(self, _chk, subprocess):
         """Test action add_osds."""
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
         self.harness._charm.peers.interface.state.joined = True
 
         action_event = MagicMock()
@@ -422,7 +422,7 @@ class TestCharm(test_utils.CharmTestCase):
     @patch("ceph.check_output")
     def test_add_osds_action_with_already_added_device_id(self, _chk, subprocess):
         """Test action add_osds."""
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
         self.harness._charm.peers.interface.state.joined = True
 
         disk = "/dev/sdb"
@@ -449,7 +449,7 @@ class TestCharm(test_utils.CharmTestCase):
     @patch("ceph.check_output")
     def test_add_osds_action_with_loop_spec(self, _chk, subprocess):
         """Test action add_osds with loop file spec."""
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
         self.harness._charm.peers.interface.state.joined = True
 
         action_event = MagicMock()
@@ -468,7 +468,7 @@ class TestCharm(test_utils.CharmTestCase):
 
     def test_add_osds_action_node_not_bootstrapped(self):
         """Test action add_osds when node not bootstrapped."""
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
 
         action_event = MagicMock()
         action_event.params = {"device-id": "/dev/sdb"}
@@ -487,7 +487,7 @@ class TestCharm(test_utils.CharmTestCase):
 
     def _test_list_disks_action(self, microceph_cmd_output, expected_disks):
         """Test action list_disks."""
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
         self.harness._charm.peers.interface.state.joined = True
 
         action_event = MagicMock()
@@ -498,7 +498,7 @@ class TestCharm(test_utils.CharmTestCase):
 
     def test_list_disks_action_node_not_bootstrapped(self):
         """Test action list_disks when node not bootstrapped."""
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
 
         action_event = MagicMock()
         self.harness.charm.storage._list_disks_action(action_event)
@@ -709,7 +709,7 @@ class TestCharm(test_utils.CharmTestCase):
 
     def test_get_rgw_endpoints_action_node_not_bootstrapped(self):
         """Test action get_rgw_endpoints when node not bootstrapped."""
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
 
         action_event = MagicMock()
         self.harness.charm.rgw._get_rgw_endpoints_action(action_event)
@@ -732,7 +732,7 @@ class TestCharm(test_utils.CharmTestCase):
         self.harness.update_config(
             {"snap-channel": "1.0/stable", "enable-rgw": "*", "namespace-projects": True}
         )
-        test_utils.add_complete_peer_relation(self.harness)
+        self.add_complete_peer_relation(self.harness)
         self.add_complete_identity_relation(self.harness)
         self.add_complete_ingress_relation(self.harness)
 

